sematic/db/models/tests/test_factories.py

```diff
@@ -91,6 +92,43 @@ def test_make_artifact():
     assert artifact.json_summary == json.dumps(json_summary, sort_keys=True)
     assert artifact.type_serialization == json.dumps(type_serialization, sort_keys=True)
 
+def test_make_artifact_with_optional_type_hint():
+    artifact = make_artifact([42], Optional[List[int]])
+
+    value_serialization = value_to_json_encodable([42], List[int])
+    type_serialization = type_to_json_encodable(List[int])
+    json_summary = get_json_encodable_summary([42], List[int])
+
+    payload = {
+        "value": value_serialization,
+        "type": type_serialization,
+        "summary": json_summary,
+    }
+
+    sha1 = hashlib.sha1(json.dumps(payload, sort_keys=True).encode("utf-8"))
+
+    assert artifact.id == sha1.hexdigest()
+    assert artifact.json_summary == json.dumps(json_summary, sort_keys=True)
+    assert artifact.type_serialization == json.dumps(type_serialization, sort_keys=True)
+
+def test_make_artifact_with_nested_union_type_hint():
+    artifact = make_artifact([42], Union[Union[List[int], str], List[str]])
```

Comment:
Nit: Do you think composing is necessary? How about: 

```suggestion
    artifact = make_artifact([42], Union[List[int], str, List[str]])
```

Reply:
Yup, the reason I kept it was to test if everything works correctly, even if the user defines nested Union. Although the chance of that happening is low.